internal/reconciler/tracepipeline/reconciler.go

```diff
@@ -137,22 +137,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	return ctrl.Result{}, r.doReconcile(ctx, &tracePipeline)
+	err = r.doReconcile(ctx, &tracePipeline)
+	if statusErr := r.updateStatus(ctx, tracePipeline.Name); statusErr != nil {
+		if err != nil {
+			err = fmt.Errorf("failed while updating status: %w: %w", statusErr, err)
+		} else {
+			err = fmt.Errorf("failed to update status: %w", statusErr)
+		}
+	}
+	return ctrl.Result{}, err
 }
 
 func (r *Reconciler) doReconcile(ctx context.Context, pipeline *telemetryv1alpha1.TracePipeline) error {
 	var err error
```

Comment:
maybe we also remove the declaration of the error here to make it consistent with the other reconcilers?

Reply:
done